build/WFCORE-3769_WFLY-10305_galleon_build.adoc

```diff
@@ -0,0 +1,166 @@
+   org.wildfly:wildfly-servlet-feature-pack used in previous releases.
+ * The default WildFly Core maven build should continue to use the legacy 
+   org.wildfly.build:wildfly-feature-pack-build-maven-plugin to produce a 'legacy
+   full feature pack'. The maven module for this should be the 'feature-pack'
+   module, and its maven coordinate should be the same
+   org.wildfly:wildfly-feature-pack used in previous releases.
+ * These legacy feature packs should be produced in such a way that their output
+   would be acceptable for release as the primary build output if this shift to 
+   Galleon had not occurred. In other words, these legacy feature packs should be
+   usable by users who are accustomed to consuming our feature packs.
+ * The WildFly Core maven build should include a profile that builds
+   new maven modules that produce 'legacy thin and fat distributions' using the legacy
+   org.wildfly.build:wildfly-server-provisioning-maven-plugin. These will use
+   the 'legacy core feature pack' discussed previously.
+ ** This maven profile should not run by default.
+ * The WildFly maven build should include a profile that builds
+   new maven modules that produce 'legacy servlet and full thin and fat distributions' 
+   using the legacy org.wildfly.build:wildfly-server-provisioning-maven-plugin. These will
+   use the 'legacy servlet feature pack' and 'legacy full feature pack' discussed previously.
+ ** This maven profile should not run by default.
+ * These legacy distributions should be produced in such a way that their output, except 
+   for their maven GAVs, would be acceptable for release as the primary build output if 
+   this shift to Galleon had not occurred. In other words, these distributions should 
+   provide a valid base for comparison to the output of the Galleon build.
+ ** The only purpose of these legacy distributions is to facilitate such comparisons.
+ * The content of the Galleon-produced distributions should functionally match the output
+   of the legacy distributions, with only minimal minor cosmetic differences.
+ * Special distributions of the server that are produced inside the WildFly Core and 
+   WildFly testsuites should be produced by Galleon, not by legacy tooling.
+ ** Copying of distribution content produced in other modules of the build into the
+    needed testsuite locations is acceptable; i.e. there is no requirement that
+    all test distributions be built using Galleon, just as there was no requirement
+    in the past to use the legacy server-provisioning plugin in all places.
+
+=== Nice-to-Have Requirements
+
+None 
+
+=== Non-Requirements
+
+ * Support for any use of the Galleon tooling outside of the maven build steps described in this document. 
+ * Long term support for the initial content details of the Galleon feature packs. The initial version
+   of these feature packs should be regarded as 'Tech Preview', i.e. subject to change in the next WildFly 
```

Comment:
Is 'Tech Preview' language used on community side or it's just product related note?

Will be Galleon feature packs (GFP) uploaded to the public maven repository? As it's just internal build tool for WF13 scope I guess no uploads of GFP.

Reply:
While they are going to be mainly used internally (by the build), possible advantages of making them public with the 'tech preview' notice is that the community and other teams could easily try using them, get some experience and give feedback on how it worked or not for them.